subsys/mpsl/mpsl_init.c

```diff
@@ -124,6 +128,171 @@ static uint8_t m_config_clock_source_get(void)
 #endif
 }
 
+#if IS_ENABLED(CONFIG_MPSL_FEM)
+
+#define MPSL_FEM_NRF21540_GPIO_POLARITY_GET(dt_property) \
+	((DT_GPIO_FLAGS(DT_NODELABEL(nrf_radio_fem), dt_property) & GPIO_ACTIVE_LOW) ? false : true)
+
+#define MPSL_FEM_GPIO_INVALID_PIN        0xFFU
+#define MPSL_FEM_GPIOTE_INVALID_CHANNEL  0xFFU
+#define MPSL_FEM_DISABLED_GPIO_CONFIG_INIT \
+	.enable       = false, \
+	.active_high  = true, \
+	.gpio_pin     = MPSL_FEM_GPIO_INVALID_PIN, \
+	.gpiote_ch_id = MPSL_FEM_GPIOTE_INVALID_CHANNEL
+
+#endif
+
+#if IS_ENABLED(CONFIG_MPSL_FEM_NRF21540_GPIO)
+static int fem_nrf21540_gpio_configure(void)
+{
+	/* FEM configuration requires gpiote and ppi channels.
+	 * Currently there is no reliable common method to dynamically
+	 * allocate such channels. FEM module needs only "some" channels
+	 * to use whichever they are, but FEM needs them for exclusive use
+	 * and does not enable them immediately.
+	 *
+	 * When common api to assign gpiote and ppi channels is available
+	 * current solution based on macros coming from Kconfig should be
+	 * reworked.
+	 */
+
+#if !DT_NODE_EXISTS(DT_NODELABEL(nrf_radio_fem))
+#error Node with label 'nrf_radio_fem' not found in the devicetree.
+#endif
+
+	mpsl_fem_nrf21540_gpio_interface_config_t cfg = {
+		.fem_config = {
+			.pa_time_gap_us  =
+				DT_PROP(DT_NODELABEL(nrf_radio_fem), tx_en_settle_time_us),
+			.lna_time_gap_us =
+				DT_PROP(DT_NODELABEL(nrf_radio_fem), rx_en_settle_time_us),
+			.pdn_settle_us   =
+				DT_PROP(DT_NODELABEL(nrf_radio_fem), pdn_settle_time_us),
+			.trx_hold_us     =
+				DT_PROP(DT_NODELABEL(nrf_radio_fem), trx_hold_time_us),
+			.pa_gain_db      =
+				CONFIG_MPSL_FEM_NRF21540_TX_GAIN_DB,
+			.lna_gain_db     =
+				CONFIG_MPSL_FEM_NRF21540_RX_GAIN_DB
+		},
+		.pa_pin_config = {
+#if DT_NODE_HAS_PROP(DT_NODELABEL(nrf_radio_fem), tx_en_gpios)
+			.enable       = true,
+			.active_high  = MPSL_FEM_NRF21540_GPIO_POLARITY_GET(tx_en_gpios),
+			.gpio_pin     = DT_GPIO_PIN(DT_NODELABEL(nrf_radio_fem), tx_en_gpios),
```

Comment:
What if the device has more than one GPIO port ?
For example:
`DT_GPIO_PIN` for `<&gpio0 12 GPIO_ACTIVE_LOW>` and `<&gpio1 12 GPIO_ACTIVE_LOW>` is 12 in both cases

Reply:
I fixed the initialization - previously it was not possible to use GPIO-1 pins, at least not in a clean way.
Now you can use pins from both `&gpio0` and `&gpio1` in any combination.